addons/sale_project/models/sale_order_line.py

```diff
@@ -93,21 +93,38 @@ def _convert_qty_company_hours(self, dest_company):
 
     def _timesheet_create_project_prepare_values(self):
         """Generate project values"""
-        account = self.order_id.analytic_account_id
-        if not account:
+        analytic_account = self.order_id.analytic_account_id
+        if not analytic_account:
             service_products = self.order_id.order_line.product_id.filtered(lambda p: p.type == 'service' and p.default_code)
             default_code = service_products.default_code if len(service_products) == 1 else None
-            self.order_id._create_analytic_account(prefix=default_code)
-            account = self.order_id.analytic_account_id
-
+            default_plan = self.env['ir.config_parameter'].with_company(self.company_id).sudo().get_param("default_analytic_plan_id_%s" % self.company_id.id)
+            if not default_plan:
+                default_plan = self.env['account.analytic.plan'].sudo().search([('company_id', '=', False)], limit=1) or \
+                               self.env['account.analytic.plan'].create({
+                                    'name': 'Default',
+                                    'company_id': False,
+                                })
+            else:
+                default_plan_id = int(default_plan)
+                default_plan = self.env['account.analytic.plan'].browse([default_plan_id])
```

Comment:
Indeed, the default plan is required. What i did before was the same as checking if the user explicilty set one in the setting. Since the default analytic plan of the setting is related to the analytic plan of the current company, What we can do instead is modify the way the analytic plan of the company is computed to always get a default plan without a company id instead when the user did not set one. I made the changes already.